### PR TITLE
[WIP] Trade logs api

### PR DESCRIPTION
**Pull request by namnm1991**

### CODE REVIEW

**Comment by favadi** [comment]
Current json response


```json
{
  "data": [
    {
      "Timestamp": "2018-10-09T07:43:42Z",
      "BlockNumber": 6481489,
      "TransactionHash": "0xaf2efc747375f6ed91a7c838d049ac053ac0fff38f84ae16293ed5b3192c23a9",
      "Index": 0,
      "EtherReceivalSender": "0x0000000000000000000000000000000000000000",
      "EtherReceivalAmount": 0,
      "UserAddress": "0x85c5c26dc2af5546341fc1988b9d178148b4838b",
      "SrcAddress": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
      "DestAddress": "0xc5bbae50781be1669306b9e001eff57a2957b09d",
      "SrcAmount": 999097000000000000,
      "DestAmount": 322802378,
      "FiatAmount": 0,
      "BurnFees": [
        {
          "ReserveAddress": "0x63825c174ab367968ec60f061753d3bbd36a0d8f",
          "Amount": 970373000000000000
        }
      ],
      "WalletFees": [
        {
          "ReserveAddress": "0x63825c174ab367968ec60f061753d3bbd36a0d8f",
          "WalletAddress": "0xc9d81352fbdb0294b091e51d774a0652ef776d99",
          "Amount": 415874000000000000
        }
      ],
      "IP": "",
      "Country": ""
    }
  ],
  "success": true
}
```

Please add `json:"name"` tag for all struct in repsonse message so they use the snake case in JSON output.

File: tradelogs/cmd/trade-logs-api/main.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	app.Action = func(c *cli.Context) error {
+		logger, err := libapp.NewLogger(c)
+		if err != nil {
+			return err
+		}
+		defer logger.Sync()
+
+		sugar := logger.Sugar()
+
+		coreClient, err := core.NewClientFromContext(sugar, c)
```
[the remark this answers is not in the document]
Comment (namnm1991) [reply tradelogs/cmd/trade-logs-api/main.go]: Ah, we need this to get token info, so we can format amount from humanized number to token wei unit.

File: tradelogs/api/http.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package api
```
[the remark this answers is not in the document]
Comment (namnm1991) [reply tradelogs/api/http.go]: ok

File: tradelogs/api/http.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/KyberNetwork/reserve-stats/tradelogs/storage"
+)
+
+// HTTPApi serve trade logs through http endpoint
+type HTTPApi struct {
```
[the remark this answers is not in the document]
Comment (namnm1991) [reply tradelogs/api/http.go]: ok, I will follow this.

File: tradelogs/api/http.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		c.JSON(
+			http.StatusOK,
+			gin.H{"success": false, "reason": fmt.Errorf("invalid to time %s", err.Error())},
+		)
+		return
+	}
+
+	c.JSON(
+		http.StatusOK,
+		gin.H{"success": true, "data": tradeLogs},
```
[the remark this answers is not in the document]
Comment (namnm1991) [reply tradelogs/api/http.go]: reserve-data return trade logs in this format {'success': true, 'data': []}, should we keep it the old way?

File: tradelogs/storage/influx_results.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	if err != nil {
+		return tradeLog, fmt.Errorf("failed to get dst_amount: %s", err)
+	}
+
+	dstAmountInWei, err := is.amountFmt.ToWei(dstAddress, humanizedDstAmount)
+	if err != nil {
+		return tradeLog, fmt.Errorf("failed to convert dst_amount: %s", err)
+	}
+
+	fiatAmount, err := influxdb.GetFloat64FromInterface(row[10])
```
[the remark this answers is not in the document]
Comment (namnm1991) [reply tradelogs/storage/influx_results.go]: ok, let's update it later

**Pull request merged**